### PR TITLE
Only stop rendering on Private routes

### PR DESCRIPTION
**Pull request by peterp**
This PR does a whole bunch, and changes some fundemental behaviour, but for the better.

Previously we would block all rendering until we were able to determine if a user was logged in, once we were able to determine that a user was logged, and they were, we would further block until we had a currentUser.

This blocking behaviour is now based on the type of route:`<Private />` blocks, `<Route />` does not. You can also supply a loading component whilst you're waiting for auth-state-hydration:

```<Private whileLoading={() => 'Loading...'} />```

Thanks to @dac09 for the idea!

/cc @Tobbe you might find this interesting too :)

### CODE REVIEW

**Comment by dac09** [comment]
This is awesome! Will test tomorrow and get back to you :). Never fail to impress me haha

**Comment by peterp** [comment]
I tested it on my side and it's working like a charm, those unauthenticated renders are hella-fast!

File: packages/auth/src/__tests__/AuthProvider.test.tsx [hunk 1]
```diff
@@ -144,6 +147,7 @@ test('Authentication flow (logged out -> login -> logged in -> logout) works as
       'currentUser: {"name":"Peter Pistorius","email":"nospam@example.net"}'
     )
   ).toBeInTheDocument()
+  expect(screen.getByText('authToken: hunter2')).toBeInTheDocument()
```
Comment (dac09) [inline packages/auth/src/__tests__/AuthProvider.test.tsx:150]: 😂

**Review by dac09 — APPROVED** [review]
Tested locally. Nice one @peterp  👍

**Comment by dac09** [comment]
`whileLoading` on individual private routes is an awesome addition, and let's us optimise the "perceived performance" even more. Really pleased, gives me some ideas on how to make react-snap rendering easier.

**Comment by dthyresson** [comment]
LGTM @dac09 let's merge and test in canary.

**Pull request merged**